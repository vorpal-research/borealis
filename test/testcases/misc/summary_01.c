#include <stdio.h>
#include <stdlib.h>

int* my_malloc(size_t size, int safe) {
    int* res = malloc(sizeof(int) * size);
    if (safe > 0) {
        if (res) return res;
        else exit(255);
    } else {
        return res;
    }
}

// int* my_malloc(size_t size) {
//     int* res = malloc(sizeof(int) * size);
//     if (res) 
//         return res;
//     else 
//         exit(255);
// }

int main() {

    int* arr = my_malloc(16, 1);
    // int* arr = my_malloc(16);

    for (int i = 0; i < 16; ++i) {
        arr[i] = i;
    }

    return arr[15];
}
