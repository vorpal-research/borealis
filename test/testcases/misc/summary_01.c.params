---summary:sum-mode:interpol
---analysis:craig-colton-bounds:off
