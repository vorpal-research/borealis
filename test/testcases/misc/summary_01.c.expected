
[
	
	{
		"location" : 
		{
			"filename" : "test/testcases/misc/summary_01.c",
			"loc" : 
			{
				"col" : 19,
				"line" : 18
			}
		},
		"type" : "BUF-01"
	},
	
	{
		"location" : 
		{
			"filename" : "test/testcases/misc/summary_01.c",
			"loc" : 
			{
				"col" : 5,
				"line" : 22
			}
		},
		"type" : "BUF-01"
	}
]
